src/main/java/org/truffleruby/language/dispatch/DispatchNode.java

```diff
@@ -128,7 +128,9 @@ public Object execute(VirtualFrame frame, Object receiver, String methodName, Ru
                     return MISSING;
                 case CALL_METHOD_MISSING:
                     // Both branches implicitly profile through lazy node creation
-                    if (metaclass == getContext().getCoreLibrary().truffleInteropForeignClass) {
+                    if (IsANode.getUncached().executeIsA(
```

Comment:
Always using an uncached node is too slow here, we need to use a cached node when possible, since this is executed on the fast path for calling `method_missing`.
Could you follow the pattern for other `@Child protected` nodes, and create this one lazily like it's done for `callForeign()`?